urllib3/exceptions.py

```diff
@@ -112,6 +112,9 @@ class ConnectTimeoutError(TimeoutError):
     "Raised when a socket timeout occurs while connecting to a server"
     pass
 
+class NewConnectionError(ConnectTimeoutError, PoolError):
```

Comment:
I can see the backward compatibility argument for `ConnectTimeoutError`, though you really want to comment that because it's definitely weird and I can see us forgetting that logic.

Otherwise I don't think `PoolError` is right: `ProtocolError` seems better to me.